sqlglot/generator.py

```diff
@@ -2591,6 +2591,15 @@ def safedpipe_sql(self, expression: exp.SafeDPipe) -> str:
     def div_sql(self, expression: exp.Div) -> str:
         return self.binary(expression, "/")
 
+    def typeddiv_sql(self, expression: exp.TypedDiv) -> str:
+        l, r = expression.left, expression.right
+
+        if l.is_type(*exp.DataType.INTEGER_TYPES) and r.is_type(*exp.DataType.INTEGER_TYPES):
+            l.replace(exp.cast(l.copy(), to=exp.DataType.Type.BIGINT))
+            r.replace(exp.cast(r.copy(), to=exp.DataType.Type.BIGINT))
```

Comment:
If I understand this correctly, e.g. `x / y` is parsed into a `TypedDiv` when reading Postgres and so when transpiling it back we just cast the x and y to ints? What about generating towards MySQL which doesn't truncate then? It doesn't seem like the casts reflect that MySQL should also truncate?

Reply:
ah, you are correct. this should be CAST(l / r AS BIGINT)